docs/Upgrade.md

```diff
@@ -1816,18 +1816,55 @@ The form components don't handle redirection anymore, as redirections are now ha
 
 To upgrade, replace all occurrences of `<FormWithRedirect>` with  `<Form>`.
 
+Also, the `render` prop has been removed, and the form content is now passed as child of the `<Form>` element.
+
 ```diff
 - import { FormWithRedirect } from 'react-admin';
 + import { Form } from 'react-admin';
 
 export const MyForm = () => (
 -    <FormWithRedirect
-+    <Form
-        render={() => ...}
-    />
+-        render={() => ...}
+-    />
++    <Form>
++        ...
++    </Form>
 );
 ```
 
+If you need to access the form state (`valid`, `invalid`, `pristine`, `dirty`), you can call [the react-hook-form `useFormState` hook](https://react-hook-form.com/api/useformstate) instead:
+
+```diff
+import { FormWithRedirect } from 'react-admin';
++ import { Form } from 'react-admin';
++ import { useFormState } from 'react-hook-form';
+
+const MyCustomForm = () => {
+    return (
+-        <FormWithRedirect
++        <Form>
+-            render={({ valid, dirty, handleSubmit }) => (
+-               <form onSubmit={handleSubmit}>
+                    ...
+-                    <SubmitButton disabled={!dirty || !valid}>Save</SubmitButton>
++                    <SubmitButton>Save</SubmitButton>
+-               </form>
+            )}
+-       />
++       </Form>
+    );
+};
+
+-const SubmitButton = ({ disabled, ...props }) => {
++const SubmitButton = (props) => {
++    const { isDirty, isValid } = useFormState();
+    return (
+-        <button disabled={disabled} {...props} />
++        <button disabled={!isDirty || !isValid} {...props} />
```

Comment:
this doesn't work - you need at least a type="submit"

Reply:
Isn't `type="submit"` the default for a button? I'll test it...

Reply:
I confirm it's working without `type="submit"`.

Do you want me to add it anyway? In this case, should I also add it to the "before" line, to make it clear that the diff is not about this?